onnx/test/inference_function_test.py

```diff
@@ -1,63 +1,73 @@
+)
+RESHAPE_SCHEMA = max(
+    (s for s in onnx.defs.get_all_schemas_with_history() if s.name == "Reshape" and s.domain == ""),
+    key=lambda s: s.since_version
+)
 
-def call_infer_types(
-    schema: onnx.defs.OpSchema, inputs: Iterable[onnx.TypeProto],
-    num_outputs: Optional[int] = None, node: Optional[onnx.NodeProto] = None
-) -> Optional[List[onnx.TypeProto]]:
-    inputs = list(inputs)
-
-    if not schema.has_type_and_shape_inference_function:  # type: ignore
-        return None
-
-    if node is None:
-        if num_outputs is None:
-            raise ValueError("Either node or num_outputs must be specified.")
-        input_names = [f"in{i}" for i in range(len(inputs))]
-        output_names = [f"out{i}" for i in range(num_outputs)]
-        node = onnx.helper.make_node(schema.name, input_names, output_names)
+_tensor = onnx.helper.make_tensor_type_proto
 
-    outputs = schema.infer_types(  # type: ignore
-        node.SerializeToString(),
-        {f"in{i}": in_type.SerializeToString() for i, in_type in enumerate(inputs)},
-        {},
-        {},
-    )
 
-    return [onnx.TypeProto.FromString(output) for output in outputs]
+def _to_tensor_types(
+        tensor_types: Dict[str, Tuple[int, Tuple[Union[int, str, None], ...]]]
+) -> Dict[str, onnx.TypeProto]:
+    return {key: onnx.helper.make_tensor_type_proto(*value) for key, value in tensor_types.items()}
 
 
-ADD_SCHEMA = max(
-    (s for s in onnx.defs.get_all_schemas_with_history() if s.name == "Add"),
-    key=lambda s: s.since_version
-)
+def _run_case(
+        schema: onnx.defs.OpSchema, input_names: List[str], output_names: List[str],
+        input_types: Dict[str, onnx.TypeProto], input_data: Optional[Dict[str, np.ndarray]] = None
+) -> Dict[str, onnx.TypeProto]:
+    if input_data is None:
+        input_data = {}
+    return onnx.shape_inference.infer_node_outputs(
+        schema,
+        onnx.helper.make_node(schema.name, input_names, output_names, domain=schema.domain),
+        input_types,
+        {key: onnx.numpy_helper.from_array(arr) for key, arr in input_data.items()}
+    )
 
 
 class TestInferenceFunctionCall(unittest.TestCase):
     def test_add_inference(self) -> None:
-        assert call_infer_types(ADD_SCHEMA, [
-            onnx.helper.make_tensor_type_proto(1, ()),
-            onnx.helper.make_tensor_type_proto(1, ())
-        ], 1) == [onnx.helper.make_tensor_type_proto(1, ())]
-        assert call_infer_types(ADD_SCHEMA, [
-            onnx.helper.make_tensor_type_proto(1, (None, 2)),
-            onnx.helper.make_tensor_type_proto(1, (2,))
-        ], 1) == [onnx.helper.make_tensor_type_proto(1, (None, 2))]
-        assert call_infer_types(ADD_SCHEMA, [
-            onnx.helper.make_tensor_type_proto(1, (None, 2)),
-            onnx.helper.make_tensor_type_proto(1, (1, 2))
-        ], 1) == [onnx.helper.make_tensor_type_proto(1, (None, 2))]
+        cases = [
+            ({'A': (1, ()), 'B': (1, ())}, {'C': (1, ())}),
```

Comment:
nit: I assume the 1 is the type encoding of float. If so, better to use the symbolic names like `TensorProto.FLOAT` instead.

Reply:
I'll fix this (and the Windows CI, looks like it's failing due to a different default int dtype for a numpy array?) tomorrow.

Reply:
I heard that the Windows CI are failing across PRs, may be unrelated to this PR. Thanks!